crates/core/tedge_mapper/src/c8y/service_monitor.rs

```diff
@@ -0,0 +1,158 @@
+use c8y_api::smartrest::topic::SMARTREST_PUBLISH_TOPIC;
+use mqtt_channel::Message;
+use mqtt_channel::Topic;
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct HealthStatus {
+    #[serde(rename = "type", default = "default_type")]
+    pub service_type: String,
+
+    #[serde(default = "default_status")]
+    pub status: String,
+}
+
+fn default_type() -> String {
+    "thin-edge.io".to_string()
+}
+
+fn default_status() -> String {
+    "down".to_string()
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct TopicInfo {
+    pub service_name: String,
+    pub child_id: Option<String>,
+}
+
+impl TopicInfo {
+    fn parse_topic_info(topic: &str) -> Self {
+        let topic_split: Vec<&str> = topic.split('/').collect();
+        let service_name = if topic_split.len() == 4 {
+            topic_split[3]
+        } else {
+            topic_split[2]
+        }
+        .to_string();
+
+        let child_id = if topic_split.len() == 4 {
+            Some(topic_split[2].to_owned())
+        } else {
+            None
+        };
+
+        Self {
+            service_name,
+            child_id,
+        }
+    }
+}
+pub fn convert_health_status_message(message: &Message, device_name: String) -> Vec<Message> {
+    let mut mqtt_messages: Vec<Message> = Vec::new();
+    let topic = message.topic.name.to_owned();
+    let topic_info = TopicInfo::parse_topic_info(&topic);
+
+    // If not Bridge health status
+    if !topic_info.service_name.contains("bridge") {
+        let payload_str = message
+            .payload_str()
+            .unwrap_or(r#""type":"thin-edge.io","status":"down""#);
+
+        let health_status = serde_json::from_str(payload_str).unwrap_or_else(|_| HealthStatus {
+            service_type: "unknown".to_string(),
+            status: "down".to_string(),
+        });
```

Comment:
To avoid having to provide twice the default values (and involuntary introducing a difference), one can chain function calls.

```suggestion
        let health_status = message
            .payload_str()
            .and_then(serde_json::from_str)
            .unwrap_or_else(|_| HealthStatus {
                service_type: "unknown".to_string(),
                status: "down".to_string(),
            });
```

Reply:
fixed